### PR TITLE
Update CAPI to v0.4.3

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What type of PR is this?**

**What this PR does / why we need it**: https://github.com/kubernetes-sigs/cluster-api/releases/tag/v0.4.3

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by CecileRobertMichon** [comment]
looks like cluster pause/unpause tests are failing 

this is the second failure in a row, I think this is an actual failure

File: internal/test/logentries/pause_events.go [hunk 1]
```diff
@@ -48,7 +48,7 @@ func GenerateCreateNotPausedLogEntries(ec EntryCriteria) []gomega.LogMatcher {
 			ec.ClusterName,
 			"msg",
 			"Cluster is not paused, allowing further processing",
-		).WithLevel(4).WithLogFunc("Info")
+		).WithLevel(6).WithLogFunc("Info")
```
Comment (CecileRobertMichon) [inline internal/test/logentries/pause_events.go:51]: https://github.com/fabriziopandini/cluster-api/commit/cf769543a93ea7f1605be10c186a5ac2997aaa3b#

**Review by mboersma — APPROVED** [review]
/lgtm

**Comment by shysank** [comment]
seeing consistent errors when deleting resource group https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/1728/pull-cluster-api-provider-azure-capi-e2e/1442612992054136832/artifacts/clusters/bootstrap/controllers/capz-controller-manager/capz-controller-manager-67bfdf96f9-9txc2/manager.log
```
I0927 22:30:08.668572       1 async.go:54] controllers/AzureCluster "msg"="long running operation is still ongoing" "azureCluster"="kcp-upgrade-uuotly" "cluster"="kcp-upgrade-uuotly" "namespace"="kcp-upgrade-p5dswv" "resource"="capz-e2e-ohyn9d" "service"="group"
I0927 22:30:08.669021       1 recorder.go:104] events "msg"="Warning"  "message"="error deleting AzureCluster kcp-upgrade-p5dswv/kcp-upgrade-uuotly: failed to delete resource group: transient reconcile error occurred: operation type DELETE on Azure resource capz-e2e-ohyn9d/capz-e2e-ohyn9d is not done. Object will be requeued after 15s" "object"={"kind":"AzureCluster","namespace":"kcp-upgrade-p5dswv","name":"kcp-upgrade-uuotly","uid":"2fa8c4c7-acfb-4091-9c05-3f568920085b","apiVersion":"infrastructure.cluster.x-k8s.io/v1alpha4","resourceVersion":"8638"} "reason"="ClusterReconcilerDeleteFailed"
I0927 22:30:08.669304       1 recorder.go:104] events "msg"="Warning"  "message"="error deleting AzureCluster kcp-upgrade-p5dswv/kcp-upgrade-uuotly: failed to delete resource group: transient reconcile error occurred: operation type DELETE on Azure resource capz-e2e-ohyn9d/capz-e2e-ohyn9d is not done. Object will be requeued after 15s" "object"={"kind":"AzureCluster","namespace":"kcp-upgrade-p5dswv","name":"kcp-upgrade-uuotly","uid":"2fa8c4c7-acfb-4091-9c05-3f568920085b","apiVersion":"infrastructure.cluster.x-k8s.io/v1alpha4","resourceVersion":"8638"} "reason"="ClusterReconcilerDeleteFailed"
E0927 22:30:08.700059       1 controller.go:304] controller/azurecluster "msg"="Reconciler error" "error"="error deleting AzureCluster kcp-upgrade-p5dswv/kcp-upgrade-uuotly: failed to delete resource group: transient reconcile error occurred: operation type DELETE on Azure resource capz-e2e-ohyn9d/capz-e2e-ohyn9d is not done. Object will be requeued after 15s" "name"="kcp-upgrade-uuotly" "namespace"="kcp-upgrade-p5dswv" "reconciler group"="infrastructure.cluster.x-k8s.io" "reconciler kind"="AzureCluster" 
I0927 22:30:08.732544       1 azurecluster_controller.go:249] controllers/AzureCluster "msg"="Reconciling AzureCluster delete" "azureCluster"="kcp-upgrade-uuotly" "cluster"="kcp-upgrade-uuotly" "namespace"="kcp-upgrade-p5dswv" 
```

**Comment by CecileRobertMichon** [comment]
The log above is expected, it just means the RG was not done deleting yet (it takes a while). The real error on that test was 

>  Message: "admission webhook \"validation.kubeadmcontrolplane.controlplane.cluster.x-k8s.io\" denied the request: KubeadmControlPlane.controlplane.cluster.x-k8s.io \"kcp-upgrade-uuotly-control-plane\" is invalid: spec.kubeadmConfigSpec.clusterConfiguration.etcd.local.dataDir: Forbidden: cannot be modified",

which is different from the MHC error we got previously.

/retest

**Pull request merged**